BHoM_UI/Components/Engine/Modify.cs

```diff
@@ -63,15 +65,52 @@ public ModifyCaller() : base()
 
         public override object Run(List<object> inputs)
         {
-            if (inputs != null && inputs.Count >= 1 && inputs[0] != null)
+            bool cloned = (inputs != null && inputs.Count >= 1 && inputs[0] != null);
+            if (cloned)
             {
                 // Deepclone must be done before the properties are set to ensure immutability
                 // TODO: DeepClone should ignore fragments and CustomData
                 inputs[0] = inputs[0].DeepClone();
             }
-            return base.Run(inputs);
+
+            object result = base.Run(inputs);
+            if (m_IsVoidOutput && cloned)
+                result = inputs[0];
+            return result;
+        }
+
+        /*************************************/
+
+        protected override void SetOutputs(MethodBase method)
+        {
+            m_IsVoidOutput = method.OutputType() == typeof(void);
+
+            if (m_IsVoidOutput)
+            {
+                ParameterInfo firstParam = method.GetParameters().FirstOrDefault();
+                if (firstParam != null)
+                {
+                    Type returnType = firstParam.ParameterType;
+                    OutputParams = new List<oM.UI.ParamInfo> { new ParamInfo {
+                        Name = "result",
+                        DataType = firstParam.ParameterType,
+                        Description = "Modified copy of the input object.",
+                        Kind = ParamKind.Output,
+                        IsRequired = true
+                    }};
```

Comment:
This seems to be overriding provided `[Output]` attributes when the method is void.

I did a local change to this, and this provided the information back from the `[Output]` attribute. However, it did then put in the message "This is a Void" - I'm guessing that needs an update in the Reflection_Engine to handle that?

That said, I recognise you probably did it this way for a reason @adecler , so just a suggested change in case this is gonna come in with an Engine update :smile:

```suggestion
                    Type returnType = firstParam.ParameterType;
                    string name = method.OutputName();
                    OutputParams = new List<ParamInfo> {
                        new ParamInfo
                        {
                            Name = (name == "") ? returnType.Name.Substring(0, 1) : name,
                            DataType = firstParam.ParameterType,
                            Description = method.OutputDescription(),
                            Kind = ParamKind.Output,
                            IsRequired = true
                        }
                    };
```

Reply:
To be honest, my logic was that all out put description would be very similar and basically a rewording of the method description. So I opted for a standard simple description to apply all the time. As we said before, having a description attribute for the sake of having one even if it adds nothing is not exactly a great feature 😄

Reply:
Ok in that case I'm happy to change this to an approval review :smile: I presume we'll still be changing the `This is a Void` in reflection though? 👼

Reply:
Can do. I have raised a separate issue in the engine: https://github.com/BHoM/BHoM_Engine/issues/2287

Reply:
Perfect :smile: do you want me to merge this when the checks are complete (assuming they don't finish before you head off) or want to wait for extra reviews?

Reply:
I ma happy for whoever to merge once all the checks have passed